src/gui/entry/EntryAttachmentsModel.cpp

```diff
@@ -65,24 +67,48 @@ int EntryAttachmentsModel::columnCount(const QModelIndex& parent) const
 {
     Q_UNUSED(parent);
 
-    return 1;
+    return Columns::ColumnsCount;
 }
 
-QVariant EntryAttachmentsModel::data(const QModelIndex& index, int role) const
+QVariant EntryAttachmentsModel::headerData(int section, Qt::Orientation orientation, int role) const
 {
-    if (!index.isValid()) {
-        return QVariant();
+    if (orientation == Qt::Horizontal && role == Qt::DisplayRole) {
+        Q_ASSERT(m_headers.size() == columnCount());
+        return m_headers[section];
     }
 
-    if (role == Qt::DisplayRole && index.column() == 0) {
-        QString key = keyByIndex(index);
+    return QAbstractListModel::headerData(section, orientation, role);
+}
 
-        return QString("%1 (%2)").arg(key,
-                Tools::humanReadableFileSize(m_entryAttachments->value(key).size()));
-    }
-    else {
+QVariant EntryAttachmentsModel::data(const QModelIndex& index, int role) const
+{
+    if (!index.isValid()) {
         return QVariant();
     }
+    const int column = index.column();
+    if (role == Qt::DisplayRole) {
+        const QString key = keyByIndex(index);
+        switch (column) {
+        case Columns::NameColumn:
```

Comment:
I would replace the inner switch/case statements with if/else if as well. Perhaps even flatten this structure to a single level, since DisplayRole && NameColumn have the same action as EditRole && NameColumn, so it can be reduced to a single condition. DisplayRole && SizeColumn and EditRole && SizeColumn are also very similar, so they could be reduced to a return guard and a another return if the previous condition wasn't met.

Reply:
Done ;)

Reply:
Much nicer and shorter. Thanks.